### PR TITLE
Replace LazyFile with file path in from_csv uses.

### PR DESCRIPTION
**Pull request by nbedi**
#596. This does cause issues with empty file tests but there may be a simple check to add and fix those.

### CODE REVIEW

**Comment by jpmckinney** [comment]
Did you make this PR without merging upstream first?

**Comment by jpmckinney** [comment]
Also, `_open_input_file` is doing some important things like substituting stdin and reading bz2 and gz files.

**Comment by nbedi** [comment]
Yup my bad, fixed my branch.

And yeah the fix won't be this easy since it can't just skip `_open_input_file`. I was just trying to see if deferring file handling to agate would fix the unicode erros.

File: csvkit/utilities/csvsort.py [hunk 1]
```diff
@@ -26,7 +26,12 @@ def main(self):
             self.print_column_names()
             return
 
-        table = agate.Table.from_csv(self.input_file, sniff_limit=self.args.sniff_limit, header=not self.args.no_header_row, column_types=self.get_column_types(), **self.reader_kwargs)
+        # Otherwise, fails with "io.UnsupportedOperation: underlying stream is not seekable".
```
Comment (jpmckinney) [inline csvkit/utilities/csvsort.py]: Hmm, I think the merge wasn't done properly as this entire if-block shouldn't be here.

**Comment by jpmckinney** [comment]
I took your approach in #640, which is merged. We still have [two PY2 bugs](https://travis-ci.org/wireservice/csvkit/jobs/144628368), which seem related to agate-excel now.

**Pull request closed without merging**